yt_dlp/extractor/txxx.py

```diff
@@ -0,0 +1,307 @@
+        'md5': '6de5bc1f13bdfc3491a77f23edb1676f',
+        'info_dict': {
+            'id': '11761',
+            'ext': 'mp4',
+            'title': 'Yui Hatano in If Yui Was My Girlfriend',
+            'uploader': 'Matheus69',
+            'duration': 3310,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://voyeurhit.com/videos/332875/charlotte-stokely-elle-alexandra-malena-morgan-lingerie/',
+        'md5': '12b4666e9c3e60dafe9182e5d12aae33',
+        'info_dict': {
+            'id': '332875',
+            'ext': 'mp4',
+            'title': 'Charlotte Stokely, Elle Alexandra, Malena Morgan-Lingerie',
+            'uploader': 'Kyle Roberts',
+            'duration': 655,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://voyeurhit.tube/videos/332875/charlotte-stokely-elle-alexandra-malena-morgan-lingerie/',
+        'md5': '12b4666e9c3e60dafe9182e5d12aae33',
+        'info_dict': {
+            'id': '332875',
+            'ext': 'mp4',
+            'title': 'Charlotte Stokely, Elle Alexandra, Malena Morgan-Lingerie',
+            'uploader': 'Kyle Roberts',
+            'duration': 655,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        group_1 = 0 if not video_id[:-3] else video_id[:-3] + '000'
+        group_2 = 0 if not video_id[:-6] else video_id[:-6] + '000000'
+        host = self._match_valid_url(url).group('host')
+        video_info_url = 'https://%s/api/json/video/86400/%s/%s/%s.json' % (host, group_2, group_1, video_id)
+        video_path_url = 'https://%s/api/videofile.php?video_id=%s&lifetime=8640000' % (host, video_id)
+
+        headers = {
+            'Referer': url,
+            'X-Requested-With': 'XMLHttpRequest',
+        }
+
+        video_info = self._download_json(video_info_url, video_id, 'Downloading video info', headers=headers)
+        if 'error' in video_info:
+            raise ExtractorError(f'Txxx said: {video_info["error"]}', expected=True, video_id=video_id)
+
+        video_file = self._download_json(video_path_url, video_id, 'Downloading video file info', headers=headers)
+        if 'error' in video_file:
+            raise ExtractorError(f'Txxx said: {video_file["error"]}', expected=True, video_id=video_id)
+
+        video_json = video_info.get('video')
+        stat_json = video_json.get('statistics')
+        user_json = video_json.get('user')
+
+        formats = []
+        for video in video_file:
+            format_id = self._get_format_id(video.get('format'))
+            formats.append(format_id)
+        quality = qualities(formats)
```

Comment:
This assumes the formats are already in order of quality. Is that true? If so, just set it directly in below loop using `enumerate`

Reply:
Not conclusive, but they do seem to be in low to high order.
Code updated.